### PR TITLE
Raise an error if the interpreter is not in the current virtualenv

### PR DESCRIPTION
**Pull request by galaunay**
# PR Summary
Following issue #1333.
Elpy now raises an error if the wanted interpreter (set by `python-shell-interpreter`) is not available within the current virtual environnement.
This avoid Elpy to just fall back to the system interpreter binaries, outside of the virtual environment, without the user noticing.

# PR checklist

- [x] Commits respect our [guidelines](../CONTRIBUTING.rst)
- [x] Tests are passing properly (see [here](https://elpy.readthedocs.io/en/latest/extending.html#running-tests) on how to run Elpy's tests)
- [x] Code is not generating new bytecode warnings

### CODE REVIEW

File: elpy-shell.el [hunk 1]
```diff
@@ -229,6 +229,11 @@ Python process. This allows the process to start up."
          (proc (get-buffer-process bufname)))
     (if proc
         proc
+      (when (and pyvenv-virtual-env
+                 (not (string-match (regexp-quote pyvenv-virtual-env)
```
Comment (jorgenschaefer) [inline elpy-shell.el]: Should this use `string-prefix-p`, and maybe `expand-file-name` for both?

**Comment by galaunay** [comment]
I also added some tests.
@jorgenschaefer Do you know if there is a way to create a temporary virtual environment to make this test more efficient ?

**Comment by jorgenschaefer** [comment]
> @jorgenschaefer Do you know if there is a way to create a temporary virtual environment to make this test more efficient ?

Creating a "real" virtualenv is a tricky process that depends on the Python version etc., so I don't think it's something we can do easily. virtualenvwrapper has a temporary virtualenv feature, but that just adds a post-deactivate hook that deletes the directory. You can either create a virtualenv for the test, or you can just set `exec-path` to something in a temporary directory and fake it that way … activating a virtualenv really is just changing `exec-path`.

**Comment by galaunay** [comment]
Ok, thanks for the clarification.
In this case I guess it is good enough to just activate an empty virtual env.

**Pull request merged**